### PR TITLE
Feature/issue 10 generate csv report

### LINKED ISSUE #10: Generate CSV report
#68

### PR DESCRIPTION
**Pull request by hydrosquall**
## What this PR does

- Fixes #10 by adding a `export_table_to_csv` function to the `DBConnector` class and corresponding unit test to `test.py`
-  Adjusts the `Config` class to enable user to toggle whether to generate CSV exports or not
- Since the `csv/` folder was in the `.gitignore`, it seemed like a reasonable place to send the CSV exports. This could be changed if you'd like.
- Adjust `README.md` + the example `Config` file to accommodate usage of the export feature

## Related thoughts

- The `test.py` suite is broken for reasons unrelated to this PR (in the `package_managers.py` class), I raised this in #66 . 
- I added a hint to where to get a Github API key in the README.md. 
- I didn't modify `CHANGELOG.md`, but could do so when this PR is approved
- The idea of sending an email attachment that I raised in my comment at the end of #10 felt out of scope for this particular PR, but perhaps it could be a future milestone.

### Checklist

(From #64). 

- [ ] Code passes all existing [tests](https://github.com/sendgrid/open-source-library-data-collector/tree/master/test). (This is unchecked because two package manager tests were broken at the time that I cloned this project, they should be fixed in a separate PR. However, I did not break any new tests).
- [x] Any new functionality added includes new unit tests in [`tests/test.py`](https://github.com/sendgrid/open-source-library-data-collector/blob/master/test/test.py)
- [x] Create or update example code to show the new functionality in action.
- [x] All code, branch, and git naming and style conventions are followed (see [`CONTRIBUTING.md`](https://github.com/sendgrid/open-source-library-data-collector/blob/master/CONTRIBUTING.md#style-guidelines--naming-conventions))
- [x] Feature branch has been rebased off of the latest `master` branch. ( see [`CONTRIBUTING.md`](https://github.com/sendgrid/open-source-library-data-collector/blob/master/CONTRIBUTING.md#creating-a-pull-request) ).

### CODE REVIEW

File: test/test.py [hunk 1]
```diff
@@ -139,5 +139,28 @@ def test_send_email(self):
...
+    def setUp(self):
+        if os.environ.get('TRAVIS') == None:
+            self.github = GitHub()
+            self.db = DBConnector()
+            self.config = Config()
+            self.github.update_library_data(self.config.github_user,
+                                            self.config.github_repos[0])
+            self.filename = "./csv/{}.csv".format(GitHubData.__tablename__)
+
+    def test_file_export_succeeds(self):
```
[the remark this answers is not in the document]
Comment (hydrosquall) [reply test/test.py]: Hi @mbernier: I could add a test to check if the CSV file has a header that matches the DB column headers, would that work?

**Comment by hydrosquall** [comment]
@mbernier - I added a test for data correctness which checks the column headers, and verifies that the export has the same number of rows as the source DB table.

**Comment by thinkingserious** [comment]
Hi @hydrosquall,

Could you please resolve the conflict? Thanks!

**Comment by hydrosquall** [comment]
@thinkingserious - attempted doing the merge conflict resolution from the Github online UI, not sure if it worked properly. Feel free to let me know if doing a merge from master wasn't the clean way to go about doing this, and I'll follow an alternate approach.

**Comment by thinkingserious** [comment]
Hi @hydrosquall,

It looks like the merge conflicts are gone 👍

**Comment by thinkingserious** [comment]
Hi @hydrosquall,

My apologies, could you please resolve the current conflicts? I tried to do it, but for some reason my changes are not sticking.

With Best Regards,

Elmer

**Comment by hydrosquall** [comment]
@thinkingserious - just took a shot at resolving the conflicts from the web UI. Hopefully that takes care of it :)

**Comment by thinkingserious** [comment]
Perfect, thank you!

**Comment by thinkingserious** [comment]
Hello @hydrosquall,<br /><br/>Thanks again for the PR!<br />                                              <br />We appreciate your contribution and look forward to                                              continued collaboration. Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**